### ISSUE TITLE
[typescript-resolvers] Some resolver query arguments are generated as optional but are required in schema

### ISSUE THREAD

**Issue by biern** [dotansimha/graphql-code-generator#2323]
**Describe the bug**
As in the title in some cases some query arguments are optional but should be required.
It looks as if the `RequireFields<...>` type is either improperly generated or should be defined in another way.

**To Reproduce**

My GraphQL schema:

```graphql
type Query {
  purchases(
    first: Int
    after: String
    orderBy: PurchaseOrderBy = creationTime
    direction: SortDirection = desc
    filter: PurchaseFilter!
  ): PurchaseConnection!
}
...
```

My `codegen.yml` config file:

```yml
generates:
  ./types/gql.ts:
    config:
      avoidOptionals: false
      maybeValue: T | undefined
    plugins:
      - add: '// tslint:disable'
      - typescript
      - typescript-resolvers
```

Following typings are generated:

```typescript
...
export type QueryPurchasesArgs = {
  first?: Maybe<Scalars['Int']>;
  after?: Maybe<Scalars['String']>;
  filter: PurchaseFilter;
  orderBy?: Maybe<PurchaseOrderBy>;
  direction?: Maybe<SortDirection>;
};

...
export type QueryResolvers = ...
  purchases?: Resolver<
    ResolversTypes['PurchaseConnection'],
    ParentType,
    ContextType,
    RequireFields<QueryPurchasesArgs, 'orderBy' | 'direction'>
  >;
```

Effectively the `purchases` query resolver `args.filter` is generated as `PurchaseFilter | undefined` due to the `RequireFields` behaviour while it shouldn't be optional based on graphql schema.

**Expected behavior**
Required arguments in query resolver are not marked as optional.

**Environment:**

- OS: linux
- `@graphql-codegen/1.5.0`: 
- NodeJS: 10.16.0

**Comment by kamilkisiela** [comment]
I tried to reproduce it but it seems fine: https://github.com/dotansimha/graphql-code-generator/pull/2347

Could you tell me if something is missing in the test? Maybe you have a reproduction?

**Comment by biern** [comment]
Hey @kamilkisiela, in my example `QueryPurchasesArgs` is generated correctly (this is what your test case cover I think). However, note that `RequireFields<QueryPurchasesArgs, 'orderBy' | 'direction'>` is applied on top of `QueryPurchasesArgs` in resolver definition. This is what makes the `filter` parameter optional.

Eventually it is the `QueryResolvers['purchases']` that is incorrect and I'm not sure how should a test case for this look like :wink:

I think that either `RequireFields` definition should be changed or `| 'filter'` added to the second argument.

**Comment by kamilkisiela** [comment]
@biern ooh, I see... I'll look at it

**Comment by kamilkisiela** [comment]
#2348 should fix it :)

**Comment by dotansimha** [comment]
Fixed in 1.6.0.
